catalog-rest-service/src/main/resources/json/schema/operations/workflows/createWorkflow.json

```diff
@@ -0,0 +1,175 @@
+      "type": "integer",
+      "default": 1
+    },
+    "startDate": {
+      "description": "Start date of the workflow",
+      "type": "string"
+    },
+    "endDate": {
+      "description": "End Date of the workflow",
+      "type": "string"
+    },
+    "workflowTimezone": {
+      "description": "Timezone in which workflow going to be scheduled.",
+      "type": "string",
+      "default": "UTC"
+    },
+    "retries": {
+      "description": "Retry workflow in case of failure",
+      "type": "integer",
+      "default": 1
+    },
+    "retryDelay": {
+      "description": "Delay between retries in seconds.",
+      "type": "integer",
+      "default": 300
+    },
+    "workflowCatchup": {
+      "description": "Workflow catchup for past executions.",
+      "type": "boolean",
+      "default": "false"
+    },
+    "scheduleInterval": {
+      "description": "Scheduler Interval for the Workflow in cron format.",
+      "type": "string"
+    },
+    "maxActiveRuns": {
+      "description": "Maximum Number of active runs.",
+      "type": "integer",
+      "default": 1
+    },
+    "workflowTimeout": {
+      "description": "Timeout for the workflow in seconds.",
+      "type": "integer",
+      "default": 60
+    },
+    "workflowDefaultView": {
+      "description": "Default view in Airflow.",
+      "type": "string",
+      "default": "tree"
+    },
+    "workflowDefaultViewOrientation": {
+      "description": "Default view Orientation in Airflow.",
+      "type": "string",
+      "default": "LR"
+    },
+    "pythonOperatorLocation": {
+      "description": "File system directory path where managed python operator files are stored.",
+      "type": "string"
+    },
+    "slaMissCallback": {
+      "description": "python method call back on SLA miss.",
+      "type": "string"
+    },
+    "onSuccessCallback": {
+      "description": "callback method on successful execution of the pipeline.",
+      "type": "string"
+    },
+    "onFailureCallback": {
+      "description": "callback method on pipeline execution failure.",
+      "type": "string"
+    },
+    "onSuccessCallbackName": {
+      "description": "python method name to call when the pipeline successfully executes.",
+      "type": "string"
+    },
+    "onSuccessCallbackFile": {
+      "description": "python file where the successful callback method is stored.",
+      "type": "string"
+    },
+    "onFailureCallbackName": {
+      "description": "python method name to call when the pipeline failed executes.",
+      "type": "string"
+    },
+    "onFailureCallbackFile": {
+      "description": "python file where the failure callback method is stored",
+      "type": "string"
+    },
+    "tasks": {
+      "description": "All the tasks that are part of pipeline.",
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/task"
+      },
+      "default": null
+    },
+    "owner": {
+      "description": "Owner of this database",
+      "type": "string"
```

Comment:
Should this be: `      "$ref": "../../type/entityReference.json",` instead of string

Reply:
I am going to remove the createWorkflow here. There needs to be refactoring a bit here.
However in here we are capturing the connection details to connect to a data source such as hive, redshift etc.. to fetch the metadata about databases, tables . so having this as reference to an entity which is not created yet would be confusing.